### ISSUE TITLE
implement syntax --timeout=INTEGER:<timeout state>

### ISSUE THREAD

**Issue by flucas1** [monitoring-plugins/monitoring-plugins#1890]
Hello,

It would be great that monitoring-plugins also impement the syntax --timeout=INTEGER:<timeout state>
as in nagios-plugins:

     -t, --timeout=INTEGER:<timeout state>
        Seconds before connection times out (default: 10)
        Optional ":<timeout state>" can be a state integer (0,1,2,3) or a state STRING

In many cases I want that a timeout is reported as state STATE_UNKNOWN, rather than CRITICAL

**Comment by RincewindsHat** [comment]
I do like that idea. It would be backwards compatible and rather helpful.

@sni @waja what do you think?

**Comment by sni** [comment]
sounds useful to me.

**Comment by waja** [comment]
Sounds like an useful enhancement.

**Comment by RincewindsHat** [comment]
referencing #2266 where something like this was implemented for `check_curl`. Probably should be generalized.

**Comment by sni** [comment]
i agree, this could be generalized whenever there is a timeout option.
